services/src/main/java/org/keycloak/models/policy/ResourcePolicyManager.java

```diff
@@ -203,29 +204,36 @@ public void processEvent(ResourcePolicyEvent event) {
+                                            policy.getId(), event.getResourceId(), event.getOperation());
+                                    policyStateProvider.scheduleAction(policy, getFirstAction(policy), event.getResourceId());
+                                } else {
+                                    // policy is not scheduled, so we run all actions immediately
+                                    log.debugf("Running all actions of policy %s for resource %s based on event %s",
+                                            policy.getId(), event.getResourceId(), event.getOperation());
+                                    KeycloakModelUtils.runJobInTransaction(session.getKeycloakSessionFactory(), session.getContext(), s -> {
+                                        getActions(policy.getId()).forEach(action -> getActionProvider(action).run(List.of(event.getResourceId())));
+                                    });
+                                }
+                            }
+                        } else {
+                            if (provider.resetOnEvent(event)) {
                                 policyStateProvider.scheduleAction(policy, getFirstAction(policy), event.getResourceId());
-                            } else {
-                                // policy is not scheduled, so we run all actions immediately
-                                log.debugf("Running all actions of policy %s for resource %s based on event %s",
-                                        policy.getId(), event.getResourceId(), event.getOperation());
-                                KeycloakModelUtils.runJobInTransaction(session.getKeycloakSessionFactory(), session.getContext(), s -> {
-                                    getActions(policy.getId()).forEach(action -> getActionProvider(action).run(List.of(event.getResourceId())));
-                                });
+                            } else if (provider.deactivateOnEvent(event)) {
+                                policyStateProvider.remove(policy.getId(), event.getResourceId());
                             }
                         }
-                    } else {
-                        if (provider.resetOnEvent(event)) {
-                            policyStateProvider.scheduleAction(policy, getFirstAction(policy), event.getResourceId());
-                        } else if (provider.deactivateOnEvent(event)) {
-                            policyStateProvider.remove(policy.getId(), event.getResourceId());
-                        }
+                    } catch (ModelValidationException e) {
+                        policy.getConfig().putSingle("enabled", "false");
+                        policy.getConfig().putSingle("validation_error", e.getMessage());
+                        updatePolicy(policy, policy.getConfig());
+                        log.debugf("Policy %s was disabled due to: %s", policy.getId(), e.getMessage());
                     }
                 });
     }
```

Comment:
I am wondering if we shouldn't do the same when running the actions - probably not needed right now because we don't have yet actions that rely on other resources (like add user to a group or role), but once we do we will probably need a similar validation mechanism or at least make the `run` method throw a validation exception if the action can't perform it's task because the referenced resource (group, role, etc) doesn't exist